lib/iinChecker.js

```diff
@@ -1,148 +1,166 @@
+        } else {
+          if ( ++providerCount < providers.length ) {
+            self.makeRequest( iin, providerCount, callback );
+          } else {
+            // Maybe one last attempt here with RegEx? If that does not work then error
+            var patterns = require( '../configs/patterns' );
+
+            // Set out null info object
+            cardInfo = {
+              iin: iin,
+              brand: null,
+              issuer: self.nullValue,
+              type: self.nullValue,
+              category: self.nullValue,
+              country: self.nullValue
+            };
+
+            _.each( patterns, function( pattern ) {
+              var regex = new RegExp( pattern.expression );
+              if ( !cardInfo.brand && regex.test( iin ) ) {
+                cardInfo.brand = self.brands[pattern.name];
+              }
+            } );
+
+            // See if RegEx worked
+            if ( cardInfo.brand && cardInfo.brand !== self.brands.UNKNOWN ) {
+              callback( null, cardInfo );
+            } else {
+              // Return the err in the error callback, or if no error send back the http status code
+              callback( error || response.statusCode );
+            }
+          }
+        }
+      } );
+    };
+
+  };
+
+  // create a shorthand then export the IinChecker object for Nodejs
+  exports = module.exports = iinChecker;
+
+  /**
+   * Lookup a card by its IIN and return a card object
+   *
+   * @param  {String} iin the number to check
+   * @param  {Function} callback function to return the card details/error
+   * @return {Object} returns the card details
+   */
+  iinChecker.prototype.lookup = function( iin, callback ) {
+    try {
+      var self = this;
+      if ( _.isUndefined( iin ) ) {
+        throw new TypeError( this.options.messages.PARAMETER_IIN_IS_UNDEFINED );
+      } else {
+        if ( _.isEmpty( iin ) ) {
+          throw new TypeError( this.options.messages.PARAMETER_IIN_IS_EMPTY );
+        } else {
+          // convert input to string incase a int is passed in
+          iin = String( iin );
+
+          // First up. Make sure we are passing a number in.
+          if ( _.isNaN( parseInt( iin, 10 ) ) ) {
+            throw new TypeError( this.options.messages.PARAMETER_IIN_IS_NOT_A_NUMBER );
+          }
+
+          // Now make sure that we are passing in 6 characters
+          if ( iin.length !== 6 ) {
+            throw new TypeError( this.options.messages.PARAMETER_IIN_IS_NOT_LONG_ENOUGH );
+          }
+
+        }
+      }
+      if ( !_.isFunction( callback ) ) {
+        throw new TypeError( this.options.messages.PARAMETER_CALLBACK_NOT_FUNCTION );
+      }
+      // if cache is turned on attempt to get card details from cache
+      if ( self.options.cache ) {
+        self.options.cache.get( iin )
+        .then( function( data ) {
+          // card details from cache
+          callback( null, data );
+        } )
+        .fail( function() {
```

Comment:
may be better to use catch() instead of fail() here as I believe more promise implementations use that and in q they are aliases for each other